warehouse/admin/views/users.py

```diff
@@ -125,3 +126,23 @@ def user_detail(request):
         return HTTPSeeOther(location=request.current_route_path())
 
     return {"user": user, "form": form, "roles": roles}
+
+
+@view_config(
+    route_name='admin.user.delete',
+    require_methods=['POST'],
+    permission='admin',
+    uses_session=True,
+    require_csrf=True,
+)
+def user_delete(request):
+    user = request.db.query(User).get(request.matchdict['user_id'])
+
+    # Delete projects one by one so they are purged from the cache
+    for project in user.projects:
+        remove_project(project, request, flash=False)
+
+    request.db.delete(user)
```

Comment:
this will implicitly cascade to their associated emails?

Reply:
also probably want a `JournalEntry`

Reply:
Yes, this will delete emails and roles for that user.

I thought about adding a `JournalEntry` here, but it seems like they're project-specific? E.g., we don't make them when a user account is created, just when stuff changes with a specific project.

Reply:
fair, I would like some record of this feature. the journal is freeform. we should keep a record that an admin used it.

Reply:
```
JournalEntry(
    submitted_by=request.user,
    action='purge_user',
    name=f'user:{user.username}),
)
```

would work and can't collide with project names.